chemicaltoolbox/rdkit/conformers.py

```diff
@@ -0,0 +1,258 @@
+    parser.add_argument('-a', '--attempts', type=int, default=0, help='number of attempts')
+    parser.add_argument('-r', '--rmsd', type=float, default=1.0, help='prune RMSD threshold')
+    parser.add_argument('-c', '--cluster', type=str.lower, choices=['none', 'rmsd', 'tfd'],
+                        help='Cluster method (RMSD or TFD). If None then no clustering')
+    parser.add_argument('-t', '--threshold', type=float,
+                        help='cluster threshold (default of 2.0 for RMSD and 0.3 for TFD)')
+    parser.add_argument('-e', '--emin', type=int, default=0,
+                        help='energy minimisation iterations (default of 0 means none)')
+    parameter_utils.add_default_io_args(parser)
+    parser.add_argument('-s', '--smiles', help='input structure as smiles (incompatible with using files or stdin for input)')
+    parser.add_argument('-f', '--outfile', type=argparse.FileType('w+'),
+                        default=sys.stdout, help="path to the result file, default it sdtout")
+    args = parser.parse_args()
+
+    if not args.threshold:
+        if args.cluster == 'tfd':
+            args.threshold = 0.3
+        elif args.cluster == 'rmsd':
+            args.threshold = 2.0
+
+    utils.log("Conformers Args: ", args)
+
+    source = "conformers.py"
+    datasetMetaProps = {"source": source, "description": "Conformer generation using RDKit " + rdBase.rdkitVersion}
+    clsMappings = {
+        "RMSToCentroid": "java.lang.Float",
+        "EnergyDelta": "java.lang.Float",
+        "EnergyAbs": "java.lang.Float",
+        "ConformerNum": "java.lang.Integer",
+        "ClusterCentroid": "java.lang.Integer",
+        "ClusterNum": "java.lang.Integer",
+        "StructureNum": "java.lang.Integer"}
+    fieldMetaProps = [
+        {"fieldName": "RMSToCentroid",
+         "values": {"source": source, "description": "RMS distance to the cluster centroid"}},
+        {"fieldName": "EnergyDelta",
+         "values": {"source": source, "description": "Energy difference to lowest energy structure"}},
+        {"fieldName": "EnergyAbs", "values": {"source": source, "description": "Absolute energy"}},
+        {"fieldName": "ConformerNum", "values": {"source": source, "description": "Conformer number"}},
+        {"fieldName": "ClusterCentroid",
+         "values": {"source": source, "description": "Conformer number of the cluster centroid"}},
+        {"fieldName": "ClusterNum", "values": {"source": source, "description": "Cluster number"}},
+        {"fieldName": "StructureNum",
+         "values": {"source": source, "description": "Structure number this conformer was generated from"}}
+    ]
+
+    if args.smiles:
+        mol = Chem.MolFromSmiles(args.smiles)
+        suppl = [mol]
+        input = None
+    else:
+        input, output, suppl, writer, output_base = rdkit_utils. \
+            default_open_input_output(args.input, args.informat, args.output,
+                                      'conformers', args.outformat,
```

Comment:
Can you also implement `args.input, args.informat, args.output, args.outformat`? Then the user should be able to upload the SMILES input as a file as well as a string. Let me know if you need help with this.